src/app/shared/services/login/login.service.ts

```diff
@@ -138,30 +143,34 @@ export class LoginService {
       );
   }
 
-  private handleLoginErrors(e) {
+  /**
+   *  Handles non descriptive errors from iam-client-lib and cache server.
+   *  TODO: think about better implementation/approach.
+   */
+  private handleLoginErrors(e, navigateOnTimeout) {
     console.error(e);
     const loginError = LOGIN_ERRORS.filter(error => e.message.includes(error.key))[0];
     if (loginError?.type === 'swal') {
-      this.displayLoginErrorWithSwal(loginError.value);
+      const config = {
+        title: 'Wrong Network',
+        text: `${loginError.value}`,
+      };
+
+      this.openSwal(config, navigateOnTimeout);
+    } else if (e.message === 'Request failed with status code 401' || e.message === 'Request failed with status code 500') {
```

Comment:
if it's possible, I think it would be better to check a `responseCode` param instead of the error message. Maybe that could be added to the error object if it isn't there already 🤔

Reply:
Ha, not really :( This error is thrown in iam-client-lib just with a message: 
![image](https://user-images.githubusercontent.com/83639710/131396402-9ec1e59d-b9c1-4990-9daf-9a09b3af4461.png)